lib/xlsx_reader/parsers/worksheet_parser.ex

```diff
@@ -250,6 +272,22 @@ defmodule XlsxReader.Parsers.WorksheetParser do
     Utils.map_attributes(attributes, @cell_attributes_mapping)
   end
 
+  defp format_cell_data(state) do
+    value = convert_current_cell_value(state)
+
+    cell_data = %Cell{
+      value: value,
+      formula: state.formula,
+      ref: state.cell_ref
+    }
+
+    case state.cell_data_format do
+      :cell -> cell_data
```

Comment:
Since this function is in the hot path, let's inline the `%Cell{...}` construction above in this branch, there's no need to allocate the struct unless we're in `:cell` mode.